### ISSUE TITLE
Release v0.30.0 Docker Image removed Journal support without notice

### ISSUE THREAD

**Issue by Histalek** [grafana/agent#2641]
After updating my grafana container image from `docker.io/grafana/agent:v0.29.0` to `docker.io/grafana/agent:v0.30.0` i'm greeted with the following warning message:

```
{"caller":"journaltargetmanager.go:29","component":"logs","level":"warn","logs_config":"<REDACTED>","msg":"WARNING!!! Journal target was configured but support for reading the systemd journal is not compiled into this build of promtail!","ts":"2022-12-20T19:10:44.168732383Z"}
```

Neither the [upgrade guide](https://grafana.com/docs/agent/v0.30/upgrade-guide/) nor the [release notes](https://github.com/grafana/agent/releases/tag/v0.30.0) mention this change at all.

Is this intended behaviour? If yes this should have been marked as a breaking change.
If no, this should be fixed.

**Comment by tpaschalis** [comment]
Hey @Histalek, thanks for the report. 👋 

No, this is not intended behavior. I think the issue is the following; for v0.30 we updated our Promtail dependency, but didn't realize that Promtail had added a [new build tag](https://github.com/grafana/loki/commit/bdb0aa0bf45ba57eb37c53c8243797d61d5762f6) named `promtail_journal_enabled` to enable support for getting journald logs, and thus this tag wasn't part of our build process.

I'll take this with the team, but I think this warrants a patch release for fixing. If the matter is super urgent, in the meantime feel free to re-compile the Agent by adding this tag, but you should hear from us soon!

**Comment by marctc** [comment]
Fixed in release v0.30.1
